### ISSUE TITLE
vmagent: custom host header no longer set when scraping

### ISSUE THREAD

**Issue by fholzer** [VictoriaMetrics/VictoriaMetrics#5969]
### Describe the bug

Until v1.97.0 it was possible to override the host header when scraping by specifying below in the job definition.
```
  - job_name: 'test'
    headers:
    - "host: host_test"
    static_configs:
      - targets: ['10.0.0.1:8082']
```

I was able to pinpoint the commit that caused this to break to f97355d9fb16877c23f8a84188fb4940d8d5637e using git bisect.

### To Reproduce

Specify a host header like shown above on vmagent version v1.96.0 and see that it is sent to the target. Do the same on v1.97.0 and see that it using the host part of the target instead of the hots header specified.

### Version

vmagent-20240313-220227-tags-v1.97.0-0-gfcc8b14f8

### Logs

not relevant

**Comment by fholzer** [comment]
I suppose the reason for this is that the standard go library's http client ignored the host header, as it is treated specially. It overwrites the host header with the host part of the URL. The only way to override this, is by setting `req.Host`. Unfortunately the documentation for this field is not very clear on this behavior. What's noteworthy is that this field also determines the server name sent as part of the TLS handshake using SNI - though as I'm not familiar with your codebase I'm not sure if you set it manually or not.

See https://pkg.go.dev/net/http?utm_source=godoc#Request.Host

**Comment by hagen1778** [comment]
Hi @fholzer! Thanks for request and pull request!
I'm not the best person for this question, but quick check of the code and relevant issues on internet suggest me that `Host`  header shouldn't be possible to override. Otherwise, it looks like security issue to me. @zekker6 @f41gh7 wdyt?

**Comment by f41gh7** [comment]
I think, it's fine to override host header for scraping case. 

 Most probably, regression caused by this change https://github.com/VictoriaMetrics/VictoriaMetrics/commit/bc7cf4950b5ff1072a33fe58c6bae1184d01068b

 fasthttp client was replaced by standard go client and fix from linked issue should help.

**Comment by valyala** [comment]
The commit b577413d3b3fa5241f2c1fac6e095ec31a7e312e should allow setting `Host` header via [`headers` option](https://docs.victoriametrics.com/sd_configs/#http-api-client-options). This commit will be included in the next release.

**Comment by valyala** [comment]
[vmagent](https://docs.victoriametrics.com/vmagent/) should properly set `Host` header in requests to scrape targets starting from [v1.100.0](https://docs.victoriametrics.com/changelog/). Closing this issue as fixed.

**Comment by valyala** [comment]
FYI, this bugfix has been included also in [v1.97.4 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.97.4).
